### DISCUSSION TITLE
Add query parameters to long URL

### DISCUSSION THREAD

**Discussion by kslstn** [YOURLS/YOURLS#3785] [Q&A]
Hey, I just updated (from a probably 2 years old version) to the current version 1.9.2. Now I can't shorten URLs with a question mark in it anymore, making it impossible to add long URLs with a query parameter.

In the console I see a 403 error:

```
Status            403
Version           HTTP/2
Transferred       367 B (199 B size)
Referrer          Policy strict-origin-when-cross-origin
Request Priority  Highest
DNS Resolution    System
```

I've checked the docs for this, but only found a mention for special characters in the short URL. Plugins seems to focus on other parts of handling query parameters. Is this a bug with my setup? What could I do to diagnose it?

**Comment by dgw** [comment]
> Is this a bug with my setup?

Seems likely. My instance of 1.9.2 shortened `https://www.google.com/?q=query+parameter&fine=true` without issue.

> What could I do to diagnose it?

Perhaps something else changed along with upgrading YOURLS, like upgrading a web server or moving to a new host? Sometimes overzealous abuse-protection filters cause this kind of thing.

**Reply by kslstn** [reply]
Alright, thanks for checking! I think you're right—I restored my previous version with which now I also experience the issue. I guess it's just been a long time ago since we added long URLs containing a `?`.

I'm on shared hosting, barely doing anything with the server so possibly it's an update I'm not aware of or did some time in the last two years that made the difference. Just to be sure I removed (renamed) the .htaccess (the only place I can think of containing custom configuration), but to no avail. For good measure I checked the server requirements, which are all met (I tried older PHP versions, switched to 8.3. MySQL is at 5.7, Apache at 2.4.61).

I've got no clue what else I can do to resolve this. I'm also surprised to experience no other issues than submitting URLs containing a `?`.

**Reply by dgw** [reply]
> I'm on shared hosting, barely doing anything with the server so possibly it's an update I'm not aware of or did some time in the last two years that made the difference.

Shared hosting is exactly the kind of place I'd expect to see overzealous abuse-protection. Many of those companies love them some `mod_security` 😸

I'd suggest to ping your hosting provider's tech support. They might be able to change rules for just your domain, and even if not you can get some insight into the real cause.
